include/nil/blueprint/comparison/comparison.hpp

```diff
@@ -156,8 +156,25 @@ namespace nil {
                 case llvm::CmpInst::ICMP_ULE:
                 case llvm::CmpInst::ICMP_SLT:
                 case llvm::CmpInst::ICMP_ULT: {
-                    using comp_component_type = components::comparison<
-                        crypto3::zk::snark::plonk_constraint_system<BlueprintFieldType>, BlueprintFieldType>;
+                    using comp_component_type = components::comparison_flag<
+                        crypto3::zk::snark::plonk_constraint_system<BlueprintFieldType>>;
+
+                    if (inst->getOperand(0)->getType()->isFieldTy()){
+                        bitness = (llvm::dyn_cast<llvm::GaloisFieldType>(inst->getOperand(0)->getType()))->getBitWidth();
+                    }
+
+                    if (bitness >= BlueprintFieldType::modulus_bits) {
```

Comment:
```
if (bitness >= (BlueprintFieldType::modulus_bits - 1)) {
      bitness = BlueprintFieldType::modulus_bits - 2;
 }
```

`bits_amount` has to be less than the field’s `modulus_bits - 1`

Reply:
Actually looks like it has to be less **or equal** to `modulus_bits - 1` 
[https://github.com/NilFoundation/zkllvm-blueprint/blob/c2c7bc85/include/nil/blueprint/components/algebra/fields/plonk/non_native/comparison_flag.hpp/#L145]
there is no difference between  `mb -1` and `mb -2` , both are able to take values between `0b00...00` and `0b111...11`, but it is not the case for `mb`, max value for which is only `0x40000000000000000000000000000000224698fc0994a8dd8c46eb2100000001` that differs from `0b111...11`